File: api/admin/security_event_log/v1/security_event_log.go
package v1

import "github.com/gogf/gf/v2/frame/g"

type SecurityEventLogListReq struct {
	g.Meta    `path:"/security-event-log/list" method:"get" tags:"安全事件" summary:"安全事件列表"`
	Page      int    `json:"page"`
	Size      int    `json:"size"`
	EventType string `json:"event_type"`
	StartTime string `json:"start_time"`
	EndTime   string `json:"end_time"`
}

type SecurityEventLogListRes struct {
	List  []*SecurityEventLogItem `json:"list"`
	Total int                     `json:"total"`
}

type SecurityEventLogItem struct {
	Id         int64  `json:"id"`
	EventType  string `json:"event_type"`
	UserId     int64  `json:"user_id"`
	Ip         string `json:"ip"`
	UserAgent  string `json:"user_agent"`
	Detail     string `json:"detail"`
	TraceId    string `json:"trace_id"`
	CreateTime string `json:"create_time"`
}

File: manage/src/api/securityEventLog.ts
import request from './request'

export interface SecurityEventLogItem {
  id: number
  event_type: string
  user_id: number
  ip: string
  user_agent: string
  detail: string
  trace_id: string
  create_time: string
}

export function fetchSecurityEventLogList(params: {
  page?: number
  size?: number
  event_type?: string
  start_time?: string
  end_time?: string
}) {
  return request.get<unknown, { data?: { list?: SecurityEventLogItem[]; total?: number } }>(
    '/admin/log/security-event-log/list',
    {
      params: {
        page: params.page ?? 1,
        size: params.size ?? 10,
        event_type: params.event_type,
        start_time: params.start_time,
        end_time: params.end_time,
      },
    }
  )
}

/** 与视图层 import 命名一致 */
export const getSecurityEventLogList = fetchSecurityEventLogList
